### PR TITLE
Added installHander

### PR DESCRIPTION
**Pull request by scruzin**
This implements `installhandler`, which handles installation requests from new devices.

New devices are provisioned in a _sandbox_ site, and the operator is informed by email.
Once the operator has completed device configuration, the device will know its client type and will be able to complete its installation.

### CODE REVIEW

File: cmd/oceancenter/main.go [hunk 1]
```diff
@@ -119,6 +154,244 @@ func (svc *service) setup(ctx context.Context) {
 		log.Fatalf("could not set up datastore: %v", err)
 	}
 	model.RegisterEntities()
+
+	// GetCreate sandbox site its admin user if it doesn't exist.
+	site, err := model.GetSite(ctx, svc.settingsStore, sandboxSite)
+	switch err {
+	case nil:
+		// Nothing to do.
+	case datastore.ErrNoSuchEntity:
```
Comment (saxon-milton) [inline cmd/oceancenter/main.go:163]: I think this should use errors.Is

File: cmd/oceancenter/main.go [hunk 2]
```diff
@@ -119,6 +154,244 @@ func (svc *service) setup(ctx context.Context) {
...
+		var owner string
```
Comment (saxon-milton) [inline cmd/oceancenter/main.go:164]: Consider simplifying this block to,
```
owner := sandboxOwner
if svc.standalone {
    owner = localEmail
}

```

File: cmd/oceancenter/main.go [hunk 3]
```diff
@@ -119,6 +154,244 @@ func (svc *service) setup(ctx context.Context) {
...
+//
+// In the case of an error, the following is returned without any
+// device configuration information. The error message is also logged.
+//
+//	er:<error-message>
+func (svc *service) installHandler(w http.ResponseWriter, r *http.Request) {
+	svc.logRequest(r)
+	ctx := r.Context()
+
+	var (
```
Comment (saxon-milton) [inline cmd/oceancenter/main.go:257]: no need for var block right ? Can do
```
wi := r.FormValue("wi")
/// etc....
```

File: cmd/oceancenter/main.go [hunk 4]
```diff
@@ -119,6 +154,244 @@ func (svc *service) setup(ctx context.Context) {
...
+
+	// Attempt to get this device.
+	dev, err := model.GetDevice(ctx, svc.settingsStore, mac)
+	if err == nil {
+		// Device already exists => inform client what we know about it.
+		writeDeviceConfig(w, dev)
+		return
+	}
+
+	if err != datastore.ErrNoSuchEntity {
```
Comment (saxon-milton) [inline cmd/oceancenter/main.go:299]: Prefer ```if !errors.Is(err,datastore.ErrNoSuchEntity)```

File: cmd/oceancenter/main.go [hunk 5]
```diff
@@ -119,6 +154,244 @@ func (svc *service) setup(ctx context.Context) {
...
+		writeError(w, http.StatusInternalServerError, fmt.Sprintf("could not send notification: %v", err))
+		return
+	}
+}
+
+// checkTOTP returns true if the supplied string is a TOTP that has
+// been generated within the given grace period, or false otherwise.
+// It is recommended to use a grace period of at least one minute.
+// TODO: Move this to utils/totp.
+func checkTOTP(s string, digits int, secret []byte, period time.Duration) (bool, error) {
```
Comment (saxon-milton) [inline cmd/oceancenter/main.go:360]: Is this a could candidate for writing a unit test ?

**Review by saxon-milton — CHANGES_REQUESTED** [review]

File: cmd/oceancenter/main.go [hunk 6]
```diff
@@ -119,6 +154,244 @@ func (svc *service) setup(ctx context.Context) {
 		log.Fatalf("could not set up datastore: %v", err)
 	}
 	model.RegisterEntities()
+
+	// GetCreate sandbox site its admin user if it doesn't exist.
```
Comment (trekhopton) [inline cmd/oceancenter/main.go:158]: *Get sandbox site and admin user, create them if they don't exist.

**Review by trekhopton — APPROVED** [review]
Minor comment but LGTM!

**Pull request merged**